File: 001.Two-Sum/two-Sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(nums) <= 1:
            return false
        other_nums = {}
        # enumerate 遍历
        for key, val in enumerate(nums):
            if target - val in other_nums:
                return [other_nums[target - val], key]
            other_nums[val] = key
        return []

    def twoSum2(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(nums) <= 1:
            return false
        k = 0
        # 遍历键名
        for i in nums:
            j = target - i
            k += 1
            temp_nums = nums[k:]
            if j in temp_nums:
                return [k - 1, nums.index(j)]
        return []


if __name__ == '__main__':
    print Solution().twoSum((1, 9, 7, 3, 7), 15)

File: 001.Two-Sum/two-Sum.js
/*
 * @Author: tsingwong 
 * @Date: 2017-10-11 18:15:50 
 * @Last Modified by: tsingwong
 * @Last Modified time: 2017-10-11 21:56:51
 */
var twoSum = function (nums, target) {
    const len = nums.length;
    if (len <= 1) {
        return [];
    }
    let other_nums = [];
    for (let [index, val] of nums.entries()) {
        const otherIndex = other_nums.indexOf(target - val);
        if (otherIndex > -1) {
            return [otherIndex, index];
        } else {
            other_nums[index] = val;
        }
    }
    return [];
};

let twoSum2 = function (nums, target) {
    const len = nums.length;
    if (len <= 1) {
        return [];
    }
    let k = 0;
    for (let val of nums) {
        let j = target - val;
        k += 1;
        let temp_nums = nums.slice(k);
        if (temp_nums.indexOf(j) > -1) {
            return [k - 1, nums.indexOf(j)];
        }
    }
    return [];
}

console.log(twoSum([1, 9, 7, 3, 7], 12));